plugins/optimization-detective/tests/test-class-od-element.php

```diff
@@ -130,22 +131,68 @@ static function ( array $schema ): array {
 		$this->assertTrue( isset( $element['customProp'] ) );
 		$this->assertTrue( $element->offsetExists( 'customProp' ) );
 
-		$this->assertEquals( $element_data, $element->jsonSerialize() );
+		// Try setting property (which is not currently supported).
+		$functions = array(
+			static function ( OD_Element $element ): void {
+				$element['isLCP'] = true;
+			},
+			static function ( OD_Element $element ): void {
+				$element->offsetSet( 'isLCP', true );
+			},
+		);
+		foreach ( $functions as $function ) {
```

Comment:
did you consider having the arrays of functions in this test come from data providers?

Reply:
That could work, but here I just wanted to test different flavors of a specific aspect of something, so adding a data provider to test everything with this one variation would I think be overkill.